### PR TITLE
[2.0.x] Attempt to solve endstops issues

### PR DESCRIPTION
**Pull request by GMagician**
This partially revert #11900 and try to solve #12366. This replace #12373

### CODE REVIEW

**Comment by GMagician** [comment]
I tested this with dual endstops and home works, also it let axis move up and down after homing.
Using M120 when one of Z endstops is pressed both Z are stopped while trying to start a Z movement with some endstop pressed such movement is aborted...

I left `update` call inside stepper because this will prevent movements when micro is already pressed and I alsways call `update` when endstops are enabled to let triggger state to be set

Hope this will also solve #12366

@ejtagle this should better fit your original endstop handling intent, correct?

File: Marlin/src/module/motion.cpp [hunk 1]
```diff
@@ -1522,6 +1522,20 @@ void homeaxis(const AxisEnum axis) {
...
+      #endif
+      #if ENABLED(Y_DUAL_ENDSTOPS)
+        case Y_AXIS:
+      #endif
+      #if Z_MULTI_ENDSTOPS
+        case Z_AXIS:
+      #endif
+      stepper.set_separate_multi_axis(false);
+      default: break;
+    }
```
Comment (thinkyhead) [inline Marlin/src/module/motion.cpp:1538]: Great! I was going to ask about this, since I noticed that `separate_multi_axis` never gets cleared.

**Comment by thinkyhead** [comment]
It looks so simple, but the best solutions usually are. The `separate_multi_axis` fix is the best part — we totally missed that somehow!

File: Marlin/src/module/motion.cpp (hunk 1 above)
Comment (GMagician) [reply Marlin/src/module/motion.cpp:1538]: It was a @ejtagle post that gave me this hint

**Pull request merged**